### DISCUSSION TITLE
How to set the top_k for knowledge base search to be greater than 10?

### DISCUSSION THREAD

**Discussion by paulwei12341** [langgenius/dify#10166] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

In my project task, I need more than 10 data points for matching and selection, but it is currently limited to 10, which is not sufficient.

### 3. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by SamJo2023** [comment]
1. Modify the value of the environment variable in ` .env ` and **docker-compose.yaml**
-  KEY `TOP_K_MAX_VALUE: ${TOP_K_MAX_VALUE:-10}` in docker-compose.yaml is changed to the maximum recall number you want
-  KEY `TOP_K_MAX_VALUE=10` in .env (directory: ./.env) Change to the maximum recall number you want
2. After the modification is completed, execute
``` cmd
docker compose down
docker compose up -d
```
3. Settings in the application:
A very critical point is the setting of position 1 in the figure below. 
It will only take effect after it is set here. Otherwise, if 2 is set greater than 10, the maximum number of returned fragments is still 10. If the values ​​set in 1 and 2 are different, the smallest value will be finally taken.

![](https://obtong-1317320194.cos.ap-guangzhou.myqcloud.com/obsidian/202509121702341.png?imageSlim)

**Reply by ooodwbooo** [reply]
same question
